lib/worker.js

```diff
@@ -68,15 +70,21 @@ Worker.prototype._getConfigUpdateAction = function() {
 Worker.prototype.stop = function() {
     var self = this;
     BaseService.prototype.stop.call(self);
+    var stoppers;
     if (Array.isArray(self.serviceReturns)) {
-        return P.each(self.serviceReturns, function(serviceRet) {
+        stoppers = P.each(self.serviceReturns, function(serviceRet) {
```

Comment:
If we expect each service to implement a graceful shutdown independently, then we should probably start all those in parallel.